cli/command/system/info.go

```
@@ -98,21 +99,15 @@ func runInfo(ctx context.Context, cmd *cobra.Command, dockerCli command.Cli, opt
 	}
 
 	if needsServerInfo(opts.format, info) {
-		if dinfo, err := dockerCli.Client().Info(ctx); err == nil {
-			info.Info = &dinfo
-		} else {
+		dinfo, err := dockerCli.Client().Info(ctx)
+		if err != nil && client.IsErrConnectionFailed(err) {
+			return cli.StatusError{StatusCode: 125, Cause: err}
```

This is probably returning too early, because we'd still be able to print the client info;


With this PR:

```bash
build/docker -H example.invalid info
error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host
```

Before this PR;

```bash
docker -H example.invalid info
Client:
 Version:    28.0.1
 Context:    default
 Debug Mode: false
 Plugins:
...

Server:
ERROR: error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host
errors pretty printing info
```

---

i see :see_no_evil: 
I'll rewrite this a bit

---

I think we may be able to use a `defer` - I gave that a quick go, and opened a PR against your PR branch; https://github.com/Benehiko/docker-cli/pull/1



Before (current v28.0.1);

```bash
docker -H example.invalid info > /dev/null
ERROR: error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host
errors pretty printing info
echo $?
1


docker -H example.invalid info --format '{{.Architecture}}' > /dev/null
error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host

echo $?
0

# invalid template
docker -H example.invalid info --format '{{.Whoops}}' > /dev/null
error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host
template: :1:2: executing "" at <.Whoops>: cant evaluate field Whoops in type system.dockerInfo
echo $?
1
```

After;

```bash
docker -H example.invalid info > /dev/null
ERROR: error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host
errors pretty printing info

echo $?
1

docker -H example.invalid info --format '{{.Architecture}}' > /dev/null
error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host

echo $?
1

docker -H example.invalid info --format '{{.Whoops}}' > /dev/null
error during connect: Get "http://example.invalid:2375/v1.48/info": dial tcp: lookup example.invalid: no such host
template: :1:2: executing "" at <.Whoops>: cant evaluate field Whoops in type system.dockerInfo

echo %?
1

```

---

I don't think we always want to return an error since we are printing errors from the server into the template. I'm guessing we only want to return the error with an exit code once the server request is made and if the server connection cannot be established. 

I've updated my PR to handle both the empty `--format` and template format. Client info will always print if no format is specified (even on server connection error).